### PR TITLE
Add Zold::PROTOCOL and expose it in the node #357

### PR DESCRIPTION
**Pull request by botanicus**
See #357

This PR:
- Adds `Zold::PROTOCOL`.
- Exposes it via `X-Zold-Protocol`.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#362](https://github.com/zold-io/zold/issues/362) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#362](https://github.com/zold-io/zold/issues/362) is assigned to @oltdaniel[/z](https://www.0crat.com/u/oltdaniel), here is [why](https://www.0crat.com/footprint/CAZPZR9FS/1880854950); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by botanicus** [comment]
@oltdaniel no headers are being tested as of now, so writing all that would require more time than allocated, so I added a puzzle instead.

**Comment by botanicus** [comment]
@yegor256 please merge.

**Comment by botanicus** [comment]
@0crat wait

**Comment by botanicus** [comment]
Pausing the time counting since the request for merge has been waiting for 2 days already.

File: lib/zold/version.rb [hunk 1]
```diff
@@ -23,5 +23,6 @@
 # Copyright:: Copyright (c) 2018 Yegor Bugayenko
 # License:: MIT
 module Zold
-  VERSION = '0.0.0'.freeze
+  VERSION  = '0.0.0'.freeze
+  PROTOCOL = '0.0.1'.freeze
```
Comment (yegor256) [inline lib/zold/version.rb]: @botanicus this is not correct. According to our White Paper, protocol is just a simple Integer. Current protocol number is `1` in all wallets.

**Comment by yegor256** [comment]
@botanicus see above. Also, we should use protocol for `Upgrades`, instead of version.

**Comment by botanicus** [comment]
@yegor256 the upgrade related tickets are not merged in yet, so I reported it in a separate ticket #369.

I fixed the protocol number, so I think it's good to be merged now.

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @oltdaniel[/z](https://www.0crat.com/u/oltdaniel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#362](https://github.com/zold-io/zold/issues/362) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256)

**Comment by ypshenychka** [comment]
@oltdaniel According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>Messages in a job [always](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html#4-avoid-noiseaddress-your-comments) start with a name of a user they are addressed to.

Please correct your [message](https://github.com/zold-io/zold/pull/362#pullrequestreview-135198629) by indicating an addressee in the beginning.

**Comment by ypshenychka** [comment]
@oltdaniel According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

No issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by ypshenychka** [comment]
@oltdaniel ping

**Comment by ypshenychka** [comment]
@oltdaniel ping

**Comment by ypshenychka** [comment]
@0crat quality bad

**Comment by 0crat** [comment]
Quality is low, no payment, see [§31](http://www.zerocracy.com/policy.html#31)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**